### DISCUSSION TITLE
RKE2 agent fails and does not start

### DISCUSSION THREAD

**Discussion by athulg93** [rancher/rke2#6490] [Q&A]
So a total newbie to RKE2 here

I have a 4 VM cluster that was working properly with 3 RKE2 server and 1 RKE2 agent.
**Situation 1:**
I tried to take a snapshot of the VM and while the snapshot was happening we tried to run the command 
"kubectl get pods -A"
and that gave the message: "the connection to the server 127.0.0.1:6443 was refused - did you specify the right port or host"
The error fixed itself after i reverted to the old snapshot and restarted all the nodes.

**Situation 2:**
Everything was working fine and i left the cluster and did not touch it.. about 3 hours later when i try it had failed again

Can you please help us understand what went wrong and how to fix it?
![image (1)](https://github.com/user-attachments/assets/ecda2b91-62b8-42bb-a6da-161906cbabe9)

**Comment by brandond** [comment]
You've provided no logs or anything else to work with- nothing but a screenshot of an error message from a client. Check the rke2-server, kubelet, and containerd logs, along with the logs under /var/log/pods.
